drivers/sensor/stm32_temp/stm32_temp.c

```diff
@@ -116,19 +154,25 @@ static int stm32_temp_init(const struct device *dev)
 	return 0;
 }
 
-#define STM32_TEMP_INST(idx)                                                                       \
-	static struct stm32_temp_data inst_##idx##_data = {                                        \
-		.adc = DEVICE_DT_GET(DT_IO_CHANNELS_CTLR(DT_INST(idx, st_stm32_temp))),            \
-		.channel = DT_IO_CHANNELS_INPUT(DT_INST(idx, st_stm32_temp))                       \
-	};                                                                                         \
-	static const struct stm32_temp_config inst_##idx##_config = {                              \
-		.avgslope = DT_INST_PROP(idx, avgslope),                                           \
-		.v25_mv = DT_INST_PROP(idx, v25),                                                  \
-		.tsv_mv = DT_INST_PROP(idx, ts_voltage_mv),                                        \
-		.is_ntc = DT_INST_PROP(idx, ntc)                                                   \
-	};                                                                                         \
-	DEVICE_DT_INST_DEFINE(idx, stm32_temp_init, NULL, &inst_##idx##_data,                      \
-			      &inst_##idx##_config, POST_KERNEL, CONFIG_SENSOR_INIT_PRIORITY,      \
-			      &stm32_temp_driver_api);
-
-DT_INST_FOREACH_STATUS_OKAY(STM32_TEMP_INST)
+static const struct stm32_temp_config stm32_temp_dev_config = {
+	.tsv_mv = DT_INST_PROP(0, ts_voltage_mv),
+#if HAS_CALIBRATION
+	.cal1_addr = (uint16_t *)DT_INST_PROP(0, ts_cal1_addr),
+	.cal2_addr = (uint16_t *)DT_INST_PROP(0, ts_cal2_addr),
+	.cal1_temp = DT_INST_PROP(0, ts_cal1_temp),
+	.cal2_temp = DT_INST_PROP(0, ts_cal2_temp),
+	.cal_vrefanalog = DT_INST_PROP(0, ts_cal_vrefanalog),
+	.cal_offset = DT_INST_PROP(0, ts_cal_offset)
+#else
+	.avgslope = DT_INST_PROP(0, avgslope),
+	.v25_mv = DT_INST_PROP(0, v25),
+	.is_ntc = DT_INST_PROP(0, ntc)
+#endif
+};
+
+static struct stm32_temp_data stm32_temp_dev_data = { .adc = DEVICE_DT_GET(
+							      DT_INST_IO_CHANNELS_CTLR(0)),
+						      .channel = DT_INST_IO_CHANNELS_INPUT(0) };
```

Comment:
Please break a line before `.adc...`, since we have plenty of space that's a shame to put all near the 100th column.

Reply:
I understand what you mean, but this is the output of clang-format with the zephyr .clang-format rules. 
I've added a "," at the end of the .channel assignment, which causes clang-format to do the better formatting.
Hope that's ok for you? Not sure this is allowed by the coding guidelines

Reply:
.clang-format rules might be broken for this case.
ok to keep the extra ',', but for next time, don't hesitate to fix things manually.
`.clang-format` rules should be considered as a helper not a rule to apply.